web/static/internal/apps/restock/restock.js

```diff
@@ -33,30 +33,52 @@ document.onkeydown = ({ key }) => {
   }
 };
 
+function valid_upc(upc) {
+    if (upc.length != 12) return false;
+    
+    check = parseInt(upc[upc.length - 1]);
+    odd_sum = 0;
+    even_sum = 0;
+    
+    for (let i = 0; i < upc.length - 1; i++) {
+        if (i % 2) {
+            even_sum += parseInt(upc[i]);
+        } else {
+            odd_sum += parseInt(upc[i]);
+        }
+    }
+    sum = (3 * odd_sum) + even_sum;
+                    
+    if ((sum + check) % 10 == 0) return true;     
+    return false;  
+};
+
 async function scan(upc) {
   report("");
 
-  try {
-    let info = await socket.request({
-      header: {
-        to: "inventory",
-        type: "info_req",
-      },
-      body: {
-        barcode: upc,
-      },
-    });
-    switch (info.header.type) {
-      case "item_info":
-        render(info.body);
-        break;
-
-      case "user_info":
-        report("Scanned user id");
-        break;
+  if(valid_upc(upc)) {
```

Comment:
In general, we should avoid putting whole bodies of functions into `if` bodies. Instead just list the preconditions and early return if necessary. Added bonus we can more easily report the issue:

```js
async function scan(upc) {
  // All commercial products should have valid UPC barcodes so a failure to validate
  // is likely caused by the scanner reading the barcode improperly.
  if (!valid_upc(upc)) {
    report("Invalid UPC detected. Try again")
    return;
  }
  
  report("")
  
  try {
    ...
```